packages/transaction-controller/src/utils/layer1-gas-fee-flow.test.ts

```diff
@@ -0,0 +1,43 @@
+import { CHAIN_IDS } from '../constants';
+import type { Layer1GasFeeFlow, TransactionMeta } from '../types';
+import { getLayer1GasFeeFlow } from './layer1-gas-fee-flow';
+
+const MOCK_UNSUPPORTED_CHAIN_ID = '0xunsupported';
+
+describe('getLayer1GasFeeFlow', () => {
+  const mockLayer1GasFeeFlows = [
+    {
+      matchesTransaction: (transactionMeta: TransactionMeta) =>
+        transactionMeta.chainId === CHAIN_IDS.OPTIMISM,
+    },
+    {
+      matchesTransaction: (transactionMeta: TransactionMeta) =>
+        transactionMeta.chainId === CHAIN_IDS.BASE,
+    },
+  ] as Layer1GasFeeFlow[];
+
+  it('should return the correct Layer1GasFeeFlow based on the transactionMeta', () => {
+    const transactionMetaOptimism = {
+      chainId: CHAIN_IDS.OPTIMISM,
+    } as unknown as TransactionMeta;
+    const transactionMetaBase = {
+      chainId: CHAIN_IDS.BASE,
+    } as unknown as TransactionMeta;
+
+    expect(
+      getLayer1GasFeeFlow(transactionMetaOptimism, mockLayer1GasFeeFlows),
+    ).toBe(mockLayer1GasFeeFlows[0]);
+    expect(
```

Comment:
Minor, but these assertions are actually better than mine in `gas-flows.test.ts` as we're validating the first match is used.

But can we make them even more stable and put a flow that doesn't match first so we ensure it skips it?